autoinstrumentation/ruby/src/autoinstrumentation.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
```

Comment:
could this code live in the ruby SDK instead?

Reply:
This would be great

Reply:
Should we make that change in the ruby auto-inst prior to merging this?